ext/zend_test/test.c

```diff
@@ -535,6 +536,15 @@ static ZEND_METHOD(ZendTestNS_Foo, method)
 	RETURN_LONG(0);
 }
 
+static ZEND_METHOD(ZendTestNS_UnlikelyCompileError, method)
+{
+	ZEND_PARSE_PARAMETERS_NONE();
+
+	return_value = ZEND_THIS;
+
+   return return_value;
```

Comment:
I suspect the equivalent of `return $this;` would rather be something like `RETURN_OBJ(Z_OBJ_P(ZEND_THIS));`, but let's ask @cmb69?

Reply:
Ah I love your solution to avoid the problem entirely :laughing:

Reply:
Yeah, I figured this works just as well.